crates/ide_assists/src/handlers/replace_if_let_with_match.rs

```diff
@@ -207,21 +207,22 @@ pub(crate) fn replace_match_with_if_let(acc: &mut Assists, ctx: &AssistContext)
         "Replace match with if let",
         target,
         move |edit| {
+            fn make_block_expr(expr: ast::Expr) -> ast::BlockExpr {
+                // unsafe blocks are parsed as BlockExpr, but are formatted without enclosing braces.
+                // If we encounter an unsafe block expr, wrap it in another BlockExpr.
+                match expr {
+                    ast::Expr::BlockExpr(block) if block.unsafe_token().is_none() => block,
```

Comment:
```suggestion
                    ast::Expr::BlockExpr(block) if block.modifier().is_none() => block,
```

Thes same problem exists with `await {}`, `try {}` and `const {}` blocks.